crates/bevy_ecs/src/world/world_cell.rs

```diff
@@ -185,9 +185,10 @@ impl<'w> WorldCell<'w> {
         let component_id = self.world.components.get_resource_id(TypeId::of::<T>())?;
         let resource_archetype = self.world.archetypes.resource();
         let archetype_component_id = resource_archetype.get_archetype_component_id(component_id)?;
+        let p = self.world.get_resource_by_id(component_id)?;
```

Comment:
Ditto here regarding perf concerns.

Reply:
IMO `WorldCell` should work on `NonSend` resources as well. I'm also way more okay with perf costs here; it's rarer, and deliberately a slower, ref-counted solution to borrow checking woes.

Reply:
It does, but it also has dedicated APIs for this, just like World, where the access checks make sense. Here it's always going to evaluate to false and just incur an unnecessary check.